doc/development/guide/architecture.rst

```diff
@@ -0,0 +1,204 @@
+backpropagation. PennyLane thus extends the automatic differentiation
+algorithms common in optimization and machine learning to include quantum and
+`hybrid computations
+<https://pennylane.ai/qml/glossary/hybrid_computation.html#backpropagation-through-hybrid-computations>`__.
+A plugin system makes the framework compatible with any gate-based quantum
+simulator or hardware.
+
+Using PennyLane, quantum computing devices (``Devices``) can be used to
+evaluate quantum nodes (``QNodes``) and to return statistics of the results. To
+process the classical information obtained, ``Interfaces`` allow using
+accelerated machine learning libraries.
+
+In most cases, computations with PennyLane are performed on a local machine
+using a simulator. Through using PennyLane plugins, one can, however, also
+utilize remote quantum devices and simulators.
+
+Components of PennyLane
+#######################
+
+.. image:: architecture_diagram.png
+    :width: 800px
+
+Devices
+*******
+
+In PennyLane, the abstraction of a quantum computation device is encompassed
+within the :class:`~.Device` class, making it one of the basic components of
+the library. It includes basic functionality that is shared for quantum
+devices, independent of the quantum information model used. PennyLane gives
+access to multiple simulators and hardware chips through its plugins; each of
+these devices is implemented as a custom class. These classes have the
+``Device`` class as their parent class.
+
+The purpose of the ``Device`` class can be summarized as:
+
+* Providing a common API to execute a quantum circuit and request
+  the measurement of the associated observable.
+* Providing an easy way of developing a new device for PennyLane
+
+Qubit based devices can use shared utilities by using the
+:class:`~.QubitDevice`.
+
+QNodes
+******
+
+A  quantum node or ``QNode`` (represented by a subclass to
+:class:`~.BaseQNode`) is an encapsulation of a function
+:math:`f(x;\theta)=R^m\rightarrow R^n` that is executed using quantum
+information processing on a quantum device.
+
+Apart from incorporating quantum functions, ``QNodes`` also offer custom
+quantum differentiation rules. Using the so-called `parameter-shift rules
+<https://pennylane.ai/qml/glossary/parameter_shift.html>`__, many quantum
+functions can be expressed through the linear combination of other quantum
+functions. As these rules allow quantum gradients to be obtained from
+``QNodes``, hybrid computations may include ``QNodes`` as part of training deep
+learnings models.
+
+PennyLane offers the following qnode types and differentiation rules:
+
+* :class:`~.QubitQNode`: qubit parameter-shift rule
+* :class:`~.CVQNode`: CV parameter-shift rule
+* :class:`~.JacobianQNode`: finite differences
+* :class:`~.DeviceJacobianQNode`: device based gradient rule
+* :class:`~.PassthruQNode`: classical backpropagation
+* :class:`~.ReversibleQNode`: reversible backpropagation
```

Comment:
@josh146 keeping this page minimalistic, I'm having second thoughts on having this list in here. With new QNodes introduced, this will also have to be extended. How about we keep the reference to the list of QNodes?